#!/usr/bin/env python
PACKAGE = "stewart"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("x", double_t, 0, "x",    0.0, -0.3,   0.3)
gen.add("y", double_t, 0, "y",    0.0, -0.3,   0.3)
gen.add("z", double_t, 0, "z",    0.25, 0.0,   0.8)
gen.add("theta", double_t, 0, "theta",    0.0, -30,   30)
gen.add("phi", double_t, 0, "phi",    0.0, -30,   30)
gen.add("psi", double_t, 0, "psi",    0.0, -30,   30)

gen.add("x_dot", double_t, 0, "x_dot",    0.0, -1.0,   1.0)
gen.add("y_dot", double_t, 0, "y_dot",    0.0, -1.0,   1.0)
gen.add("z_dot", double_t, 0, "z_dot",    0.0, -1.0,   1.0)
gen.add("theta_dot", double_t, 0, "theta_dot",    0.0, -300,   300)
gen.add("phi_dot", double_t, 0, "phi_dot",    0.0, -300,   300)
gen.add("psi_dot", double_t, 0, "psi_dot",    0.0, -300,   300)

exit(gen.generate(PACKAGE, "stewart", "TestSetpoint"))